### PR TITLE
🚨✨ Source Convex - switch the JSON format requested from Convex

### PR DESCRIPTION
**Pull request by nipunn1313**
## What

We're updating the json format which is requested from the backend from `convex_json` to `json`. The new format is more readable/usable (rather than prioritizing roundtrippable).

## How
Updating the source connector to request `json` format (rather than `convex_json` format). The format is a tad simpler and cleaner in how it handles a few cases.

Added a `fmt` parameter to the configuration structure, defaulting it to `json` format. Intentionally left it out of the yml.

We updated the airbyte-client version number which we send up as a header so the server can detect the new version.

## Recommended reading order
1. There are two commits. First commit updates formatting / fixes mypy. No behavior change.
2. Second commit: `source.py` - Updates json format
3. Second commit: `unit tests`

## 🚨 User Impact 🚨

The users will see a simpler JSON format which encodes a few values a bit differently. Most exports will not see a difference.
Examples:
Bigints go from `{"$integer": "55"}` to `"55"`
Infinite floats go from `{"$float": "Inf"}` to `{"Inf"}`
... a few more cases

## Pre-merge Actions

<summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

### CODE REVIEW

**Comment by nipunn1313** [comment]
Hi. I am blocked on this - in order to make sure all the tests and formatting pass CI.
image

**Review by marcosmarxm — DISMISSED** [review]
Thanks for the contribution @nipunn1313 and sorry the missing update here.
This introduces a breaking change to existing connectors? If someone update the message read by Airbyte will be different?
Convex is deprecating the `convex_json` or you're introducing only another supported type?
If the first, what is the ETA to remove it?

Can you remove the `parse_config` function? The UI already apply this validation and it's duplicating features here.
If you have only a format type you as a constant in the code, if users can select the format type, which I don't believe it brings any performance to them add the option in the spec.json/yaml file.

**Comment by nipunn1313** [comment]
> This introduces a breaking change to existing connectors? 

Yes... sort of. It (subtly/minorly) simplifies the JSON format for subsequent documents.

Fortunately, we have stats on who is using the airbyte connector with convex and it is only one customer - whom we are directly in contact with. They are ok with this change.

> If someone update the message read by Airbyte will be different? Convex is deprecating the `convex_json` or you're introducing only another supported type?

There is only one documented type `json`. https://docs.convex.dev/database/types
The backend will continue to support the old `convex_json` format indefinitely, but it is not documented. Again, we only have one customer using airbyte with whom we are in close contact with.

> Can you remove the `parse_config` function? The UI already apply this validation and it's duplicating features here. If you have only a format type you as a constant in the code, if users can select the format type, which I don't believe it brings any performance to them add the option in the spec.json/yaml file.

Yes - I can do that.

**Comment by nipunn1313** [comment]
Hello. I've removed `parse_config` and wanted to check in here.

**Comment by nipunn1313** [comment]
Hello. I wanted to check in here again.

**Comment by marcosmarxm** [comment]
@nipunn1313 at the moment the team is dedicated reviewing Hackathon contribution. We're going to return to the community backlog in the future.

**Comment by nipunn1313** [comment]
Hello. Wanted to check in here. I understand your team has been busy recently.

What expectations should I have around the timing of this review? (IE - when should I check in again?).

**Comment by nataliekwong** [comment]
Hey @nipunn1313 , I apologize for the delay here. This is certainly not the experience we want you to have when contributing. Your PR, among others, sparked conversations internally about how we can improve the speed of PR review for those contributing their own company's connectors. 

@katmarkham is our steward of API sources here, and is working with her team to set a process to fast-track these in the future. For this particular PR, the team is hoping to start review by the end of next week as other projects are getting shuffled.

File: airbyte-integrations/connectors/source-convex/source_convex/source.py [hunk 1]
```diff
@@ -31,39 +45,45 @@
...
+CONVEX_CLIENT_VERSION = "0.3.0"
 
 
 # Source
 class SourceConvex(AbstractSource):
     def _json_schemas(self, config: ConvexConfig) -> requests.Response:
         deployment_url = config["deployment_url"]
         access_key = config["access_key"]
-        url = f"{deployment_url}/api/json_schemas?deltaSchema=true&format=convex_json"
+        url = f"{deployment_url}/api/json_schemas?deltaSchema=true&format=json"
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-convex/source_convex/source.py:56]: Look like braking changes as schema will be updated.

**Review by lazebnyi — APPROVED** [review]
Approved with condition. The braking changes process must be followed.

**Review by lazebnyi — APPROVED** [review]
Approved with condition. The braking changes process must be followed.

**Comment by nataliekwong** [comment]
@lazebnyi In the comments, it's noted there is only 1 user of the connector and they are already aware of this change as the contributor is already in close contact with the user. While it is a schema change, I recommend not requiring the breaking change process given we can simply inform the user of the connector of the change. @katmarkham

**Comment by nipunn1313** [comment]
> While it is a schema change, I recommend not requiring the breaking change process given we can simply inform the user of the connector of the change.

Yep! That sounds good to us.

**Review by marcosmarxm — APPROVED** [review]
Releasing without breaking change as @nataliekwong commented.

**Comment by marcosmarxm** [comment]
@nipunn1313 thanks for the contribution. Waiting the master branch allow to merge your contribution :)

**Pull request merged**